src/libs/SearchQueryUtils.ts

```diff
@@ -228,10 +226,15 @@ function getQueryHash(query: SearchQueryJSON): number {
                 }
                 return -1;
             });
-            orderedQuery += ` ${buildFilterValuesString(key, sortedFilterValues ?? [])}`;
+            orderedQuery += `${buildFilterValuesString(key, sortedFilterValues ?? [])}`;
         });
+    const recentSearchHash = UserUtils.hashText(orderedQuery, 2 ** 32);
```

Comment:
Thinking more about it. What if a user explicitly types policyID:X sortOrder:2? Should we instead just be hashing the input string instead of parts of the AST? That'd ensure anything the user types differently would be a new recent search

Reply:
We cannot hash string input. My argument: when user types **policyID:X sortOrder:asc** and **sortOrder:asc policyID:X**, we will get different hashes even though the query results will be the same.

I think we should exclude policyID like @rayane-djouah said. It may create some confusion when user types policy:X into query, but:
1. We have a policy switcher, which is dedicated way of changing policyID, so typing policyID into SearchRouter query is the worst way to change your policyID
2. We don't even autocomplete policyID key, so how will a user know that such option exists.
3. It's unlikely that a real user will type something like **policyID:31294891892741873**, without the help of autocomplete.

What's more if user selects recentSearch with not the policyID he wanted, he can easily switch it on SearchResults page with policy switcher and it's not such a big problem(just like he switches sortOrder or sortBy). WDYT @rayane-djouah @luacmartins

Reply:
I think we can go with your current solution for now, but:

>when user types policyID:X sortOrder:asc and sortOrder:asc policyID:X

TBH I'd be ok to show both of these as different recent searches, even though they are the same search at the end of the day. My argument is that the user explicitly typed both of those, so they'd probably expect to see both in the recent search area.

>We don't even autocomplete policyID key, so how will a user know that such option exists.

I think we should introduce policyID autocomplete. Maybe we need a new `workspace:` filter for that though, so the autocomplete shows the policy name instead of ID.

cc @JmillsExpensify @trjExpensify for your thoughts.